src/qilib/utils/storage/mongo.py

```diff
@@ -277,15 +284,23 @@ def update_individual_data(self, data: Any, tag: TagType, field: Union[str, int]
                                  self._encode_field(self._serialize(field)))
 
     def get_latest_subtag(self, tag: TagType) -> Optional[TagType]:
-        child_tags = sorted(self.list_data_subtags(tag))
+        child_tags = self.list_data_subtags(tag, document_limit=1)
         if len(child_tags) == 0:
             return None
 
         return tag + [child_tags[-1]]
 
-    def list_data_subtags(self, tag: TagType) -> TagType:
+    def list_data_subtags(self, tag: TagType, document_limit: int = 0) -> TagType:
```

Comment:
In the unit tests test_storage_mongo.py (test_list_subtags), list_data_subtags is never tested with document_limit other than the default value. For example something like:

        self.storage.save_data('test', ['hello', 'world'])
        self.storage.save_data('test', ['hello', 'planet'])
        self.storage.save_data('test', ['hello', 'universe'])
        results = self.storage.list_data_subtags(['hello'], 2)
        self.assertEqual(results, ['world', 'universe'])  #descending

Reply:
Updated the unit test test_list_subtags to check for limit parameter